sbin/pfdhcplistener

```diff
@@ -42,61 +43,50 @@ use pf::config qw(
     %ConfigNetworks
     @inline_enforcement_nets
 );
+use pf::file_paths qw($var_dir);
 use pf::util;
 use pf::config::util;
 use pf::services::util;
 use pf::util::dhcp;
 use pf::StatsD;
-use List::MoreUtils qw(any);
+use List::MoreUtils qw(any uniq);
 use NetAddr::IP;
-use pf::SwitchFactory;
 use MIME::Base64();
 use NetPacket::Ethernet qw(ETH_TYPE_IP ETH_TYPE_IPv6);
 use NetPacket::IP;
 use NetPacket::IPv6;
 use NetPacket::UDP;
 use pf::CHI::Request;
 
-
-pf::SwitchFactory->preloadConfiguredModules();
-
 # initialization
 # --------------
 # assign process name (see #1464)
-our $PROGRAM_NAME = "pfdhcplistener";
-
-my $logger = get_logger( $PROGRAM_NAME );
-
-# init signal handlers
-POSIX::sigaction(
-    &POSIX::SIGHUP,
-    POSIX::SigAction->new(
-        'reload_config', POSIX::SigSet->new(), &POSIX::SA_NODEFER
-    )
-) or $logger->logdie("pfdhcplistener: could not set SIGHUP handler: $!");
-
-POSIX::sigaction(
-    &POSIX::SIGTERM,
-    POSIX::SigAction->new(
-        'normal_sighandler', POSIX::SigSet->new(), &POSIX::SA_NODEFER
-    )
-) or $logger->logdie("pfdhcplistener: could not set SIGTERM handler: $!");
-
-POSIX::sigaction(
-    &POSIX::SIGINT,
-    POSIX::SigAction->new(
-        'normal_sighandler', POSIX::SigSet->new(), &POSIX::SA_NODEFER
-    )
-) or $logger->logdie("pfdhcplistener: could not set SIGINT handler: $!");
-
+our @REGISTERED_TASKS;
+our $IS_CHILD = 0;
+our $PARENT_PID;
+our %CHILDREN;
+our @TASKS_RUN;
+our $ALARM_RECV = 0;
+our $running = 1;
+
+$SIG{HUP}  = \&reload_config;
+$SIG{INT}  = \&normal_sighandler;
+$SIG{TERM} = \&normal_sighandler;
+$SIG{CHLD} = \&child_sighandler;
+$SIG{ALRM} = \&alarm_sighandler;
+
+$SIG{PIPE} = 'IGNORE';
 
 my %args;
-getopts( 'dhi:', \%args );
+getopts( 'dh', \%args );
```

Comment:
Don't believe we should, it can come in handy at some point, especially in testing and when in 5 years we get zammitsd as a new init system which will make us go back to daemonized processes, we'll be ready

So lesson here, always be ready for zammitsd